packages/panna-sdk/src/react/components/collectibles/processing-step.tsx

```diff
@@ -0,0 +1,113 @@
+import { Loader2Icon } from 'lucide-react';
+import { useEffect, useRef } from 'react';
+import { UseFormReturn } from 'react-hook-form';
+import {
+  liskSepolia,
+  prepareContractCall,
+  PrepareContractCallResult,
+  sendTransaction,
+  TokenERC
+} from 'src/core';
+import { Abi } from 'viem';
+import { useActiveAccount, usePanna } from '@/hooks';
+import { getEnvironmentChain } from '@/utils';
+import { Address, PreparedTransaction } from '../../../core/types/external';
+import { DialogHeader, DialogTitle } from '../ui/dialog';
+import { useDialogStepper } from '../ui/dialog-stepper';
+import { Typography } from '../ui/typography';
+import { SendCollectibleFormData } from './schema';
+
+const DEFAULT_ERC1155_VALUE = 1;
+const DEFAULT_ERC1155_DATA = '0x';
+
+type ProcessingStepProps = {
+  form: UseFormReturn<SendCollectibleFormData>;
+};
+
+export function ProcessingStep({ form }: ProcessingStepProps) {
+  const { client, chainId } = usePanna();
+  const { next, goToStep, lastStep } = useDialogStepper();
+  const account = useActiveAccount();
+  const initializeTokenSend = useRef(true);
+  const { collectible, token, recipientAddress } = form.getValues();
+
+  // TODO: Possibly create hook for this logic
+  useEffect(() => {
+    let transaction: PrepareContractCallResult;
+    const chain = getEnvironmentChain(chainId);
+    if (token.type === TokenERC.ERC721) {
+      transaction = prepareContractCall({
+        client,
+        chain,
+        method:
+          'function safeTransferFrom(address from, address to, uint256 tokenId)',
+        params: [
+          account?.address as Address,
+          recipientAddress as Address,
+          BigInt(collectible.id)
+        ],
+        address: token.address as Address
+      });
+    } else {
```

Comment:
Please consider adding a final `else` block to throw error for any unidentifiable collectible types.
```suggestion
    } else if (token.type === TokenERC.ERC1155) {
```